Assets/MRTK/Providers/WindowsMixedReality/XR2018/WindowsMixedRealityDeviceManager.cs

```diff
@@ -754,9 +836,90 @@ public void RemoveControllerForSuspendWorkaround(InteractionSource interactionSo
             RemoveController(interactionSource);
         }
 
-        #endregion Controller Utilities
 
-        #region Unity InteractionManager Events
+#if HP_CONTROLLER_ENABLED
+        private void AddController(object sender, MotionController e)
+        {
+            lock (trackedMotionControllerStates)
+            {
+                Handedness controllingHand;
+                switch (e.Handedness)
+                {
+                    default:
+                        controllingHand = Handedness.None;
+                        break;
+                    case MotionControllerHandedness.Left:
+                        controllingHand = Handedness.Left;
+                        break;
+                    case MotionControllerHandedness.Right:
+                        controllingHand = Handedness.Right;
+                        break;
+                }
+
+                IMixedRealityPointer[] pointers = RequestPointers(SupportedControllerType.HPMotionController, controllingHand);
+                InputSourceType inputSourceType = InputSourceType.Controller;
+
+                string nameModifier = controllingHand == Handedness.None ? inputSourceType.ToString() : controllingHand.ToString();
+                var inputSource = Service?.RequestNewGenericInputSource($"HP Motion Controller {nameModifier}", pointers, inputSourceType);
+
+                HPMotionController detectedController;
+                detectedController = new HPMotionController(TrackingState.NotTracked, controllingHand, inputSource);
```

Comment:
How does this behave? Or, I guess:

1. Do non-HP motion controllers come through this API?
1. Does this reconcile with HP motion controllers that come through the Unity APIs. or do we end up with one `WindowsMixedRealityMotionController` from there and an additional `HPMotionController` from here?

Reply:
My thought was we'd need to add `HPMotionController` to this section:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/cfa683f472c774964c48eb2d4194f36a0aee5458/Assets/MRTK/Providers/WindowsMixedReality/XR2018/WindowsMixedRealityDeviceManager.cs#L752-L789

gated on the existence of the touchpad or something (since it doesn't have one). That way, we're only creating controllers in one spot, and we could react to this event just by saving off the new `MotionControllerState`.

Reply:
So a `MotionControllerState` needs a corresponding `MotionController` to actively fetch data from which from my knowledge is only accessible via the events fired by  `MotionControllerWatcher`. which is why we have the following

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/cfa683f472c774964c48eb2d4194f36a0aee5458/Assets/MRTK/Providers/WindowsMixedReality/XR2018/WindowsMixedRealityDeviceManager.cs#L362-L365

I guess we can condition on the creation of an HPMotionController based primarily on the existence of the corresponding key in `trackedMotionControllers` (presuming that event got fired first), and then the lack of touchpad second.